util/chplenv/chpl_cpu.py

```diff
@@ -193,6 +193,17 @@ def get(flag, map_to_compiler=False, get_lcd=False):
 
     return cpu_tuple(argname or 'none', cpu or 'unknown')
 
+@memoize
+def get_llvm_target_cpu():
```

Comment:
It looks to me like the `get` function with `map_to_compiler=True`, when `compiler_val` is one of `'llvm', 'gnu', 'clang'`, is doing the same `cpu_llvm_synonyms` lookup as the new `get_llvm_target_cpu` function does regardless of compiler. I get why the new function would be helpful in the case of a compiler besides one of those, but I see that one of the failing configurations was with `gnu` compiler, failing with the same issue. So I think I must be missing something -- why would we be seeing this failure with the `gnu` compiler as well?

Reply:
We compute the flags for running `clang` in order to handle extern blocks & do so even if we are generating C for another compiler. That was the original issue.